components/compliance-service/reporting/relaxting/reports.go

```diff
@@ -557,6 +558,144 @@ func (backend *ES2Backend) GetReport(reportId string, filters map[string][]strin
 	return report, errorutils.ProcessNotFound(nil, reportId)
 }
 
+// GetNodeReport returns header information about a single report
+func (backend *ES2Backend) GetNodeReport(reportId string, filters map[string][]string) (*reportingapi.Report, error) {
+	var report *reportingapi.Report
+	depth, err := backend.NewDepth(filters, false)
+	if err != nil {
+		return report, errors.Wrap(err, "GetNodeReport unable to get depth level for report")
+	}
+	queryInfo := depth.getQueryInfo()
+
+	//normally, we compute the boolQuery when we create a new Depth obj.. here we, instead call a variation of the full
+	// query builder. we need to do this because this variation of filter query provides this func with deeper
+	// information about the report being retrieved.
+	queryInfo.filtQuery = backend.getFiltersQueryForDeepReport(reportId, filters)
+
+	logrus.Debugf("GetNodeReport will retrieve report %s based on filters %+v", reportId, filters)
+
+	fsc := elastic.NewFetchSourceContext(true).Include(
+		"node_uuid",
+		"node_name",
+		"environment",
+		"roles",
+		"platform",
+		"profiles",
+		"end_time",
+		"status",
+		"status_message",
+		"version")
+
+	if queryInfo.level != ReportLevel {
+		fsc.Exclude("profiles")
+	}
+	logrus.Debugf("GetNodeReport for reportid=%s, filters=%+v", reportId, filters)
+
+	searchSource := elastic.NewSearchSource().
+		FetchSourceContext(fsc).
+		Query(queryInfo.filtQuery).
+		Size(1)
+
+	source, err := searchSource.Source()
+	if err != nil {
+		return report, errors.Wrap(err, "GetNodeReport unable to get Source")
+	}
+	LogQueryPartMin(queryInfo.esIndex, source, "GetNodeReport query searchSource")
+
+	searchResult, err := queryInfo.client.Search().
+		SearchSource(searchSource).
+		Index(queryInfo.esIndex).
+		FilterPath(
+			"took",
+			"hits.total",
+			"hits.hits._source",
+		).
+		Do(context.Background())
+
+	if err != nil {
+		return report, errors.Wrap(err, "GetNodeReport unable to complete search")
+	}
+
+	logrus.Debugf("GetNodeReport got %d reports in %d milliseconds\n", searchResult.TotalHits(),
+		searchResult.TookInMillis)
+
+	// we should only receive one value
+	if searchResult.TotalHits() > 0 && searchResult.Hits.TotalHits > 0 {
```

Comment:
will there be any case where searchResult.TotalHits() > 1 or  searchResult.Hits.TotalHits > 1
if now can we add that condition for better control?